### PR TITLE
Implement storage events

### PR DESCRIPTION
**Pull request by christothes**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Review by KrzysztofCwalina — APPROVED** [review]

**Review by jsquire — APPROVED** [review]
CodeOwners changes look good.   I don't have deeper insight into the provisioning libraries, but nothing jumps out at me in the general sense.

File: sdk/provisioning/Azure.Provisioning.CloudMachine/src/OFX/StorageFile.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    public override bool Equals(object obj) => base.Equals(obj);
+
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public override int GetHashCode() => base.GetHashCode();
+
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public override string ToString() => $"{Path}";
+
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public new Type GetType => base.GetType();
```
Comment (tg-msft) [inline sdk/provisioning/Azure.Provisioning.CloudMachine/src/OFX/StorageFile.cs:67]: Did you mean to turn this into a property?

File: sdk/provisioning/Azure.Provisioning.CloudMachine/src/OFX/StorageFile.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    /// <summary>
+    ///
+    /// </summary>
+    /// <param name="result"></param>
+    /// <remarks>returns null if the file is not created as a return value of a service method call.</remarks>
+    public static implicit operator Response?(StorageFile result) => result._response;
+
+    public CancellationToken CancellationToken { get; internal set; }
+
+    public BinaryData Download()
```
Comment (tg-msft) [inline sdk/provisioning/Azure.Provisioning.CloudMachine/src/OFX/StorageFile.cs:31]: Should this be async? My naive impression was most things in CM were async by default.

**Review by tg-msft — APPROVED** [review]
The changes look good, including CODEOWNERS since CM is living in here for now.

File: sdk/provisioning/Azure.Provisioning.CloudMachine/src/OFX/StorageFile.cs (hunk 2 above)
Comment (christothes) [reply sdk/provisioning/Azure.Provisioning.CloudMachine/src/OFX/StorageFile.cs:31]: will consider this for later

File: sdk/provisioning/Azure.Provisioning.CloudMachine/src/OFX/StorageFile.cs (hunk 1 above)
Comment (christothes) [reply sdk/provisioning/Azure.Provisioning.CloudMachine/src/OFX/StorageFile.cs:67]: This came over from the old CM code base, so I don't recall now, but fixed it anyway :)

**Pull request merged**